docker/nginx/conf.d/server/server.local

```diff
@@ -0,0 +1,34 @@
+include /etc/nginx/conf.d/include/init-optional-variables;
+
+location /skynet/blocklist {
+    content_by_lua_block {        
+        local httpc = require("resty.http").new()
+
+        ngx.req.read_body() -- ensure the post body data is read before using get_body_data
+
+        -- proxy blocklist update request
+        -- 10.10.10.10 points to sia service (alias not available when using resty-http)
+        local res, err = httpc:request_uri("http://10.10.10.10:9980/skynet/blocklist", {
+            method = "POST",
+            body = ngx.req.get_body_data(),
+            headers = {
+                ["Content-Type"] = "application/x-www-form-urlencoded",
+                ["Authorization"] = require("skynet.utils").authorization_header(),
+                ["User-Agent"] = "Sia-Agent",
+            }
+        })
+
+        -- print error and exit with 500 or exit with response if status is not 204
+        if err or (res and res.status ~= ngx.HTTP_NO_CONTENT) then
+            ngx.status = (err and ngx.HTTP_INTERNAL_SERVER_ERROR) or res.status
+            ngx.header["content-type"] = "text/plain"
+            ngx.say(err or res.body)
+            return ngx.exit(ngx.status)
+        end
+
+        require("skynet.blocklist").reload()
```

Comment:
Are we reloading the entire blocklist each time we block a new skylink? Can we instead have an `add` method that does `x.shared.blocklist:set(hash, true)` and `ngx.shared.blocklist:flush_expired()`?

Reply:
Reloading whole blocklist is really fast and we do that very rarely - tops couple of times a day. Otherwise I would need to handle resolver skylinks in the script since blocklist automatically resolves them and blocks underlying skylink v1. The improvement I would be looking for is a native skyd endpoint that would accept a skylink and return whether it's blocked or not - discussed on discord.

Reply:
I thought we already had the V1 skylink.

Anyway, if this only happens rarely then we can leave it.